### PR TITLE
Fix handling of single/paired_interleaved inputs

### PR DESCRIPTION
**Pull request by pvanheus**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by pvanheus** [comment]
The previous version of the snippy wrapper re-used the `fastq_input` tag name for both a `conditional` and a `param` within that conditional, which caused errors when the relevant option was selected. This patch changes the names for single and interleaved paired inputs and adds a test case for the single fastq input.

File: tools/snippy/snippy.xml [hunk 1]
```diff
@@ -200,6 +199,16 @@
             <output name="snpgff" ftype="gff3" file="b_fna_ref_mincov_2_minqual_60.snps.gff" />
         </test>
 
+        <test> <!-- test 3 - fasta ref one snp single -->
+            <param name="ref" value="reference.fasta" ftype="fasta" />
+            <param name="fastq_input_selector" value="single" />
+	    <param name="fastq_input_single" value="b_2.fastq" ftype="fastqsanger" />
```
Comment (bgruening) [inline tools/snippy/snippy.xml]: indentation

**Comment by bgruening** [comment]
Looks good to me. However, I wonder why this is a problem. Scoping should prevent such errors. Do you have the error message around?

Thanks for adding a test!!!

**Comment by pvanheus** [comment]
> Looks good to me. However, I wonder why this is a problem. Scoping should prevent such errors. Do you have the error message around?
> 
> Thanks for adding a test!!!

The error is:

```
Traceback (most recent call last):
  File "/tmp/tmpLywvdI/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 218, in prepare_job
    job_wrapper.prepare()
  File "/tmp/tmpLywvdI/galaxy-dev/lib/galaxy/jobs/__init__.py", line 870, in prepare
    self.command_line, self.extra_filenames, self.environment_variables = tool_evaluator.build()
  File "/tmp/tmpLywvdI/galaxy-dev/lib/galaxy/tools/evaluation.py", line 455, in build
    raise e
NotFound: cannot find 'fastq_input_selector' while searching for 'fastq_input.fastq_input_selector'
```
Which to me suggests that somehow re-using the name `fastq_input` masks the original `fastq_input` somehow. This should probably be considered a bug in Galaxy.

File: tools/snippy/snippy.xml [hunk 2]
```diff
@@ -88,13 +87,13 @@
                 <param name="fastq_input2" type="data" format="fastqsanger,fasta" label="Select second set of reads" help="Specify dataset with reverse reads"/>
             </when>
             <when value="single">
-                <param name="fastq_input" type="data" format="fastqsanger,fasta" label="Select fastq dataset" help="Specify dataset with single reads"/>
+                <param name="fastq_input_single" type="data" format="fastqsanger,fasta" label="Select fastq dataset" help="Specify dataset with single reads"/>
             </when>
             <when value="paired_collection">
                 <param name="fastq_input" format="fastqsanger,fasta" type="data_collection" collection_type="paired" label="Select a paired collection" help="See help section for an explanation of dataset collections"/>
```
Comment (nsoranzo) [inline tools/snippy/snippy.xml]: Need to fix also this `fastq_input`, I believe.

Comment (pvanheus) [reply tools/snippy/snippy.xml]: fixed in the latest version I've committed.

**Comment by bgruening** [comment]
@pvanheus if you have time, please feel free to add a minimal test tool to Galaxy (or enhance one) to demonstrate this is a bug. Thanks for all your contributions!

File: tools/snippy/snippy.xml (hunk 2 above)
Comment (nsoranzo) [reply tools/snippy/snippy.xml]: @pvanheus I don't see it, I think this should be renamed `fastq_input_collection` or something like that.

**Comment by pvanheus** [comment]
Ah the collection input doesn't trigger the bug. I'll work on a test case
to try and explore what is going on in more detail.

**Comment by nsoranzo** [comment]
> Which to me suggests that somehow re-using the name fastq_input masks the original fastq_input somehow. This should probably be considered a bug in Galaxy.

@pvanheus Was this error happening during `planemo test` or when running it through the Galaxy UI?

**Comment by pvanheus** [comment]
In planemo serve

**Pull request merged**